numpy/ma/core.py

```diff
@@ -5869,6 +5869,18 @@ def __getitem__(self, indx):
 
         """
         m = self._mask
+        if m[indx].size > 1:
```

Comment:
@ahaldane True, my reasoning was that the status quo does not fail for subarrays of size 1.  So there is a trade-off between changing existing behaviour, or being consistent with new behaviour.